### PR TITLE
Passkey prompt: Replace headlessui with own dialog

### PR DESCRIPTION
**Pull request by jmderby**
## Description

Replace headlessui and use our existing dialog component for all passkey prompts

_small bug fix in demo during edge case when wallet fails to create, we show a sign out button to prevent deadstate_

Screenshot 2025-03-21 at 6 08 40 PM

## Test plan

passkey prompts look stunning and function as expected

## Package updates

@crossmint/client-sdk-react-ui

### CODE REVIEW

File: packages/client/ui/react-ui/src/components/auth/PasskeyPrompt.tsx [hunk 1]
```diff
@@ -246,13 +228,29 @@ export function PasskeyPrompt({ state, appearance }: PasskeyPromptProps) {
...
-    padding: "0.875rem",
-    width: "100%",
-    backgroundColor: appearance?.colors?.buttonBackground ?? "#04AA6D",
-    color: appearance?.colors?.textPrimary ?? "white",
-    borderRadius: appearance?.borderRadius ?? "8px",
-    borderColor: appearance?.colors?.border ?? "#04AA6D",
-    borderWidth: "1px",
-    fontWeight: "bold",
-});
+const PrimaryButton = ({
```
Comment (jmderby) [inline packages/client/ui/react-ui/src/components/auth/PasskeyPrompt.tsx:231]: is a copy pasta from our existing auth buttons so styling stays consistent

File: apps/wallets/smart-wallet/next/src/app/_lib/providers.tsx [hunk 2]
```diff
@@ -30,7 +30,7 @@ function CrossmintProviders({ children }: { children: ReactNode }) {
                     defaultChain: walletType === "evm-smart-wallet" ? "polygon-amoy" : undefined,
                 }}
                 appearance={{
-                    borderRadius: "16px",
+                    borderRadius: "24px",
```
Comment (jmderby) [inline apps/wallets/smart-wallet/next/src/app/_lib/providers.tsx:33]: gives a more polished rounded vibe and suits the demo styles already

**Review by AlbertoElias — APPROVED** [review]
